src/ui/server/controllers/insurance/your-buyer/failed-to-pay-on-time/index.ts

```diff
@@ -0,0 +1,160 @@
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.render} Failed to pay on time page
+ */
+export const get = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const {
+      referenceNumber,
+      buyer: { buyerTradingHistory },
+    } = application;
+
+    return res.render(TEMPLATE, {
+      ...singleInputPageVariables({
+        ...pageVariables(referenceNumber),
+        BACK_LINK: req.headers.referer,
+        HTML_FLAGS,
+      }),
+      ...pageVariables(referenceNumber),
+      userName: getUserNameFromSession(req.session.user),
+      applicationAnswer: buyerTradingHistory[FIELD_ID],
+    });
+  } catch (error) {
+    console.error('Error getting failed to pay on time %O', error);
+
+    return res.redirect(PROBLEM_WITH_SERVICE);
+  }
+};
+
+/**
+ * post
+ * Check failed to pay on time validation errors and if successful, redirect to the next part of the flow.
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} Next part of the flow or error page
+ */
+export const post = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const { referenceNumber, migratedV1toV2, totalContractValueOverThreshold } = application;
+
+    const payload = constructPayload(req.body, [FIELD_ID]);
+
+    const validationErrors = generateValidationErrors(payload);
+
+    if (validationErrors) {
+      return res.render(TEMPLATE, {
+        ...singleInputPageVariables({
+          ...pageVariables(referenceNumber),
+          BACK_LINK: req.headers.referer,
+          HTML_FLAGS,
+        }),
+        ...pageVariables(referenceNumber),
+        userName: getUserNameFromSession(req.session.user),
+        submittedValues: payload,
+        validationErrors,
+      });
+    }
+
+    const saveResponse = await mapAndSave.buyerTradingHistory(payload, application);
+
+    if (!saveResponse) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    /**
+     * If is a change route
+     * redirect to CHECK_YOUR_ANSWERS
+     */
+    if (isChangeRoute(req.originalUrl)) {
+      return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`);
+    }
+
+    /**
+     * If is a check-and-change route
+     * redirect to CHECK_AND_CHANGE_ROUTE
+     */
+    if (isCheckAndChangeRoute(req.originalUrl)) {
+      return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_AND_CHANGE_ROUTE}`);
+    }
+
+    /**
+     * if totalContractValue is over the threshold
+     * redirect to CREDIT_INSURANCE_COVER
+     * otherwise it should redirect to the BUYER_FINANCIAL_INFORMATION page
+     */
+    if (totalContractValueOverThreshold || migratedV1toV2) {
```

Comment:
We can remove `migratedV1toV2` since this is a new feature/change for all V2 applications